### PR TITLE
LibDiff+patch: Add support for --strip, --directory, and add some tests

### PR DESCRIPTION
**Pull request by shannonbooth**
Two more options taking steps to more useful patch implementation

Next steps after this, supporting multiple patches in the same patch file, and support adding a file from scratch.

### CODE REVIEW

**Comment by shannonbooth** [comment]
Thanks!

File: Userland/Utilities/patch.cpp [hunk 1]
```diff
@@ -29,9 +29,15 @@ static ErrorOr<void> do_patch(StringView path_of_file_to_patch, Diff::Patch cons
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
+    StringView directory;
+
     Core::ArgsParser args_parser;
+    args_parser.add_option(directory, "Change to the working directory to <directory> before applying the patch file", "directory", 'd', "directory");
```
Comment (AtkinsSJ) [inline Userland/Utilities/patch.cpp:36]: Should this be "Change the working directory to..."?

File: Userland/Utilities/patch.cpp [hunk 2]
```diff
@@ -30,9 +30,11 @@ static ErrorOr<void> do_patch(StringView path_of_file_to_patch, Diff::Patch cons
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
     StringView directory;
+    Optional<size_t> strip_count;
 
     Core::ArgsParser args_parser;
     args_parser.add_option(directory, "Change to the working directory to <directory> before applying the patch file", "directory", 'd', "directory");
+    args_parser.add_option(strip_count, "Strip given of leading path components from file names (defaults as basename)", "strip", 'p', "count");
```
Comment (AtkinsSJ) [inline Userland/Utilities/patch.cpp:37]: Should this be "String given number of..."?

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Code seems fine, just a couple of confusing sentences.

**Comment by shannonbooth** [comment]
pushed rebase now CI fix is merged in

**Review by ADKaster — APPROVED** [review]

**Pull request merged**